src/pybind/mgr/rbd_support/module.py

```diff
@@ -483,13 +502,802 @@ def get_perf_counters(self, pool_spec, sort_by):
+                TASK_REF_POOL_NAMESPACE: image_spec[1],
+                TASK_REF_IMAGE_NAME: image_spec[2]}
+        task = self.find_task(refs)
+        if task:
+            return 0, task.to_json(), ''
+
+        with self.open_ioctx(image_spec) as ioctx:
+            self.validate_image_migrating(ioctx, image_spec)
+
+            status = rbd.RBD().migration_status(ioctx, image_spec[2])
+            if status['state'] != rbd.RBD_IMAGE_MIGRATION_STATE_EXECUTED:
+                raise rbd.InvalidArgument("Image {} has not completed migration".format(
+                    self.format_image_spec(image_spec)), errno=errno.EINVAL)
+
+            return 0, self.add_task(ioctx,
+                                    "Committing image migration for {}".format(
+                                        self.format_image_spec(image_spec)),
+                                    refs), ''
+
+    def queue_migration_abort(self, image_spec):
+        image_spec = self.extract_image_spec(image_spec)
+        self.log.info("queue_migration_abort: {}".format(image_spec))
+
+        refs = {TASK_REF_ACTION: TASK_REF_ACTION_MIGRATION_ABORT,
+                TASK_REF_POOL_NAME: image_spec[0],
+                TASK_REF_POOL_NAMESPACE: image_spec[1],
+                TASK_REF_IMAGE_NAME: image_spec[2]}
+        task = self.find_task(refs)
+        if task:
+            return 0, task.to_json(), ''
+
+        with self.open_ioctx(image_spec) as ioctx:
+            self.validate_image_migrating(ioctx, image_spec)
+
+            return 0, self.add_task(ioctx,
+                                    "Aborting image migration for {}".format(
+                                        self.format_image_spec(image_spec)),
+                                    refs), ''
+
+    def task_cancel(self, task_id):
+        self.log.info("task_cancel: {}".format(task_id))
+
+        if task_id not in self.tasks_by_id:
+            self.log.debug("tasks: {}".format(str(self.tasks_by_id)))
+            raise KeyError("No such task {}".format(task_id))
+
+        task = self.tasks_by_id[task_id]
+        task.cancel()
+
+        remove_in_memory = True
+        if self.in_progress_task and self.in_progress_task.task_id == task_id:
+            self.log.info("Attempting to cancel in-progress task: {}".format(str(self.in_progress_task)))
+            remove_in_memory = False
+
+        # complete any associated event in the progress module
+        self.complete_progress(task)
+
+        # remove from rbd_task omap
+        with self.open_ioctx((task.refs[TASK_REF_POOL_NAME],
+                              task.refs[TASK_REF_POOL_NAMESPACE])) as ioctx:
+            self.remove_task(ioctx, task, remove_in_memory)
+
+        return 0, "", ""
+
+    def task_list(self, task_id):
+        self.log.info("task_list: {}".format(task_id))
+
+        if task_id:
+            if task_id not in self.tasks_by_id:
+                raise KeyError("No such task {}".format(task_id))
```

Comment:
Should we rise an exception here? Because it logs like below:
```
2019-07-26T11:10:57.208+0300 7fb047494700  0 log_channel(audit) log [DBG] : from='client.4694 -' entity='client.admin' cmd=[{"prefix": "rbd task list", "target": ["mgr", ""], "task_id": "e87e6984-8953-400d-8860-31dfd22764c8"}]: dispatch
2019-07-26T11:10:57.208+0300 7fb047c95700  0 mgr[rbd_support] Fatal runtime error: 'No such task e87e6984-8953-400d-8860-31dfd22764c8'
Traceback (most recent call last):
  File "/home/mgolub/ceph/ceph.ci/src/pybind/mgr/rbd_support/module.py", line 1284, in handle_command
    return self.task.handle_command(inbuf, prefix[9:], cmd)
  File "/home/mgolub/ceph/ceph.ci/src/pybind/mgr/rbd_support/module.py", line 1191, in handle_command
    return self.task_list(cmd.get('task_id'))
  File "/home/mgolub/ceph/ceph.ci/src/pybind/mgr/rbd_support/module.py", line 1163, in task_list
    raise KeyError("No such task {}".format(task_id))
KeyError: 'No such task e87e6984-8953-400d-8860-31dfd22764c8'
```
which does not look very nice. I am not sure it even should be logged as an error, I would log it as debug.

The same is probably for `task_cancel`.

Reply:
Updated